pkg/types/vsphere/platform.go

```diff
@@ -0,0 +1,5 @@
+package vsphere
+
+// Platform stores any global configuration used for vsphere
+// platforms.
+type Platform struct{}
```

Comment:
I think just empty platform may not be super useful even in UPI installations. I would like this to report some common values such as Datastore, datacenter etc. cc @dav1x

Reply:
> I think just empty platform may not be super useful even in UPI installations. I would like this to report some common values such as Datastore, datacenter etc. cc @dav1x

`report some common values such as Datastore, datacenter etc` the install-config does not `report` and is used for `configuring`.

Reply:
yes, sorry wrong choice of words. in cluster-storage-operaror we receive install-config as configmap and in case of vsphere we will receive almost no useful information..

Reply:
all clusteroperators need to be configured using `config.openshift.io` https://github.com/openshift/api/tree/master/config/v1 as their top level config.

The installer team is working to deprecate `cluster-config-v1` configmap as upgrades will be difficult if all operators do not use https://github.com/openshift/api/tree/master/config/v1

what kind of information is the storage operator looking for?
- is it more than the cloudproviderconfig for vsphere platform?

    this is going to be available through https://github.com/openshift/api/pull/245

- are some storage options storage related only?

    for UPI installations users will be required to setup storage by setting up cluster-storage-operator config object. 
    eg: https://github.com/openshift/installer/pull/1416/files#diff-1a64e7030769df4d25b1c09ddb8f62deR287 (edited)

Reply:
So if I understand correctly:

1. for UPI - CVO will not install `cluster-storage-operator`  and user must configure `cluster-storage-operator` manually?
2. For general IPI - I assuming CVO must create a instance of StorageOperatorConfig object(doesn't exist yet)  before starting cluster-storage-operator?  I will look around.

Reply:
At least in its current iteration - cluster-storage-operator does not require any configuration at all. All it does is, based on cloudprovider platform it installs right storageclass. 

For AWS - we didn't need any information from cloudprovider (other than the fact that, we are on AWS) but for vsphere we at minimum need datastore being used.  I think https://github.com/openshift/api/pull/245 would for us and cluster storage operator can watch for `Infrastructure` type and update storageclass being installed. Presumably after derek's PR, `Infrastructure` type will have cloudprovider config and hence vsphere datastore name.

This of course can change in future. For example, if user wants to use local storage or something else.

Reply:
yes, I have an issue created for moving from cluster-config-v1 to `Infrastructure` https://github.com/openshift/cluster-storage-operator/issues/18. cluster-config-v1 was the only way to determine the platform until 2 weeks ago. The storage operator only needs to know what platform openshift is installed on. It does/should not care if it is UPI or IPI, if it recognizes what is in `Infrastructure` it will create a default StorageClass. IMO there is no need for a separate object unless/until we expand the scope of the storage operator

Reply:
> So if I understand correctly:
> 
>     1. for UPI - CVO will not install `cluster-storage-operator`  and user must configure `cluster-storage-operator` manually?

CVO always installs an operator.

>     2. For general IPI - I assuming CVO must create a instance of StorageOperatorConfig object(doesn't exist yet)  before starting cluster-storage-operator?  I will look around.

CVO does not create instances for operator config.